src/controllers/payment/bacsProof.controller.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+const Constants = require('../../constants')
+const Dynamics = require('../../dynamics')
+const BaseController = require('../base.controller')
+const Configuration = require('../../persistence/entities/configuration.entity')
+const Payment = require('../../persistence/entities/payment.entity')
+const StandardRuleType = require('../../persistence/entities/standardRuleType.entity')
+const RecoveryService = require('../../services/recovery.service')
+
+const parseAmount = (amount) => {
+  if (amount) {
+    return Number.parseFloat(amount.replace(/[^0-9.]/g, ''))
+  }
+}
+
+module.exports = class BacsProofController extends BaseController {
+  async doGet (request, h, errors) {
+    const pageContext = this.createPageContext(h, errors)
+    const context = await RecoveryService.createApplicationContext(h, { standardRule: true })
+    const { application, standardRule } = context
+
+    const bacsPayment = await Payment.getBacsPayment(context)
+    if (!bacsPayment) {
+      throw new Error('Bacs payment details not found')
+    }
+
+    // Check if this an MCP permit as this changes the payment reference
+    const { categoryName } = await StandardRuleType.getById(context, standardRule.standardRuleTypeId)
+    const isMcp = Constants.MCP_CATEGORY_NAMES.find((mcpCategoryName) => mcpCategoryName === categoryName)
+
+    pageContext.bacs = {
+      paymentReference: `${isMcp ? 'MCP' : 'WP'}-${application.applicationNumber}`,
+      amount: bacsPayment.value.toLocaleString(),
+      sortCode: Constants.BankAccountDetails.SORT_CODE,
+      accountNumber: Constants.BankAccountDetails.ACCOUNT_NUMBER,
+      accountName: Constants.BankAccountDetails.ACCOUNT_NAME,
+      ibanNumber: Constants.BankAccountDetails.IBAN_NUMBER,
+      swiftNumber: Constants.BankAccountDetails.SWIFT_NUMBER,
+      paymentEmail: await Configuration.getValue(context, Dynamics.BACS_EMAIL_CONFIG)
+    }
+
+    if (request.payload) {
+      pageContext.formValues = request.payload
+    } else {
+      const { customerPaymentReference = '', customerPaymentAmount = '', customerPaymentDate } = bacsPayment
+      const [year, month, day] = customerPaymentDate ? customerPaymentDate.toISOString().substring(0, 10).split('-') : ['', '', '']
```

Comment:
Possibly don't need toISOString here if you set the "isDate" property in the mapping for the "customerPaymentDate" field in the "payment" entity.  See the "submittedOn" field in the "application" entity for a reference.

Reply:
I have indeed used isDate and that does the conversion when posting to Dynamics but when reading (as here) it converts the value into a Date object so I need to convert it (back) into a string to display the component parts on the page.